src/d3d12.zig

```diff
@@ -1004,8 +1425,12 @@ pub const Resource = struct {
     }
 
     pub fn deinit(resource: *Resource) void {
-        const d3d_resource = resource.d3d_resource;
-        _ = d3d_resource.lpVtbl.*.Release.?(d3d_resource);
+        if (resource.mem_allocator) |mem_allocator| {
+            mem_allocator.destroyResource(resource.*) catch {};
+        } else {
+            // if this is a committed resource (no allocation), we can just free it
```

Comment:
what is the benefit of having the dedicated allocator and committed resources?

Reply:
I found this [stackexchange post](https://gamedev.stackexchange.com/questions/165435/why-are-committed-resources-better-than-placed-resources) that has two different views; according to the comments on the original post it's preferred to leave the decision to the driver, but the second post states the opposite.